src/plugins/liveobjects/objectid.ts

```diff
@@ -0,0 +1,35 @@
+import type BaseClient from 'common/lib/client/baseclient';
+
+export type LiveObjectType = 'map' | 'counter';
+
+/**
+ * Represents a parsed object id.
+ *
+ * @internal
+ */
+export class ObjectId {
+  private constructor(
+    readonly type: LiveObjectType,
+    readonly hash: string,
+  ) {}
+
+  /**
+   * Create ObjectId instance from hashed object id string.
+   */
+  static fromString(client: BaseClient, objectId: string | null | undefined): ObjectId {
+    if (client.Utils.isNil(objectId)) {
+      throw new client.ErrorInfo('Invalid object id string', 50000, 500);
+    }
+
+    const [type, hash] = objectId.split(':');
+    if (!type || !hash) {
+      throw new client.ErrorInfo('Invalid object id string', 50000, 500);
+    }
+
+    if (!['map', 'counter'].includes(type)) {
+      throw new client.ErrorInfo(`Invalid object type in object id: ${objectId}`, 50000, 500);
```

Comment:
not something to change in this PR, but we're using a lot of `50000` error codes here, are they all definitely internal errors rather than client errors? either way would be good to make sure we're using unique codes for some of these so that we can configure docs help pages

Reply:
yea, everything so far comes from the server and expected to work as is, so it's at least a 5** class codes.
once we start adding write API, we should start seeing client errors.

> either way would be good to make sure we're using unique codes for some of these so that we can configure docs help pages

agree, I'll create a separate ticket for this, so we will go through them and align with what realtime has and update docs if needed.

Reply:
https://ably.atlassian.net/browse/DTP-1011